_data-prepper/pipelines/configuration/sources/s3.md

```diff
@@ -43,25 +84,70 @@ Option | Required | Type | Description
 
-## pipeline
+Option | Required | Type    | Description
+:--- | :--- |:--------| :---
+skip_lines | No | Integer | The number of lines to skip before creating events. You can use this configuration to skip common header rows. Defaults to `0`.
+header_destination | No | String  | A key value to assign the header line of the S3 object. If this option is specified then each Event will contain a header_destination field.
 
-Source for reading from another pipeline.
+### json
+
+The `json` option parses each S3 object as a single JSON object from a JSON array, then creates a Data Prepper log event for each object in the array.
+
+### csv
+
+The `csv` option parses object from a comma-seperated value (CSV) format, with each row producing a Data Prepper log event.
+
+Option | Required | Type   | Description
+:--- |:---------|:------------| :---
+delimiter | Yes | Integer  | The delimiter separating columns. Defaults to `,`.
+quote_character | Yes  | String  | The character used as a text qualifier for CSV data. Defaults to double quote (`"`).
+header | No   | String List | The header containing the column names used to parse CSV data.
+detect_header | No   | Boolean   | Whether the first line of the S3 object should be interpreted as a header. Defaults to `true`.
+
+## Using `s3_select` with the `s3` source
+
+When configuring `s3_select` to parse S3 objects, use the following options.
+
+
+Option | Required  | Type        | Description
+:--- |:-----------------------|:------------| :---
+expression | Yes, when using `s3_select` | String | The expression used to query the object. Maps directly to the [Expression](https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html#AmazonS3-SelectObjectContent-request-Expression) property.
+expression_type | No  | String | The type of the provided expression. The default value is `SQL`. Maps directly to the [ExpressionType](https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html#AmazonS3-SelectObjectContent-request-ExpressionType)
+input_serialization | Yes, when using `s3_select` | String  | Provides the S3 Select file format. Amazon S3 uses this format to parse object data into records and returns only records that match the specified SQL expression. Maybe be `csv`, `json`, or `parquet`.
+compression_type | No | String  | Specifies object's compression format. Maps directly to the [CompressionType](https://docs.aws.amazon.com/AmazonS3/latest/API/API_InputSerialization.html#AmazonS3-Type-InputSerialization-CompressionType)
+csv | No | [csv](#s3_select_csv) | Provides the CSV configuration for processing CSV data.
+json | No | [json](#s3_select_json) | Provides the JSON configuration for processing JSON data.
+
+### csv<a name="s3_select_csv"></a>
```

Comment:
Was the code in this heading intentional?

Reply:
Yes. It sets a custom header reference so that Jekyll knows what `#csv` header to go when there's multiple `csv` headers.